Update ShouldProcess for PowerArubaSW

blurpiflurp opened the pull request:
Hi!

This proposed change adds support for -confirm:$false instead of using internal logic and "-noconfirm" flags.

However you might want to work a bit with the message displayed if its not to your liking.

```powershell
PS > Remove-ArubaSWVLAN -id 10

Confirm
Are you sure you want to perform this action?
Performing the operation "Remove VLAN" on target "Vlan ID 10".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): n
PS > Remove-ArubaSWVLAN -id 10 -Confirm:$false
PS >
```

Commit:
Vlans: Use ShouldProcess (replace -noconfirm)

Hi!

This proposed change adds support for -confirm:$false instead of using internal logic and "-noconfirm" flags.

However you might want to work a bit with the message displayed if its not to your liking.

PS > Remove-ArubaSwitchVLAN -id 10

Confirm
Are you sure you want to perform this action?
Performing the operation "Remove VLAN" on target "Vlan ID 10".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): n
PS > Remove-ArubaSwitchVLAN -id 10 -Confirm:$false
PS >

Commit:
Update Vlans.Tests.ps1

Commit:
Update Vlans.ps1

blurpiflurp commented:
I've also updated the vlan tests and also the help section of the cmdlet.

alagoutte commented:
Hi,

yes, i need to use ShouldProcess for all (remove) command !

What do you have make change on Connection.ps1 ?

blurpiflurp commented:
It should have been a separate pull.

I changed it so it uses write-verbose if you specify -verbose instead of having “noverbose” flag on it.

And I also added so the name of the switch is added to the connection so if for instance you have 100 switches you can do
$arubaswitchcollection|where name -eq 'switch1' or when adding vlans you get which switch you're adding it to.

I've made a small function that imports vlans from an excel file and adds them to a collection of switches. When doing this its preferable that errors etc refers to a specific switch. Or for instance see this code that I use for cleaning up a lab I was doing.

```powershell
function Remove-ArubaLabVLAN {
    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
    Param(
        [Parameter (Mandatory = $False)]
        [ValidateNotNullOrEmpty()]
        [PSObject]$connection = $DefaultArubaSWConnection
    )
    Begin {
    }

    Process {
        if ($null -eq $connection -or [string]::IsNullOrEmpty($connection)) {
            write-error "No ArubaSwitch specified"
            return
        }
        if ($connection.count -gt 1 -and $connection.GetType().basetype.name -eq 'Array') {
            $buff = foreach ($aconn in $connection) {
                Remove-ArubaLabVLAN -connection $aconn
            }
            $buff
        } else {
            Write-Progress -activity "Remove Vlan on $($connection.Name)" -id 4 -percentcomplete 0
            $tot = 8
            foreach ($id in @(601..608)) {
                $count = $id-600
                $uri = "rest/v4/vlans/${id}"
                $target = "{0} on {1}" -f $id, $connection.Name
                if ($PSCmdlet.ShouldProcess($target, "Remove VLAN")) {
                    Write-Progress -activity "Remove Vlan on $($connection.Name)" -status $id -id 4 -percentcomplete $($count/$tot*100)
                    $null = Invoke-ArubaSWWebRequest -method "DELETE" -uri $uri -connection $connection
                    Write-Progress -activity "Remove Vlan on $($connection.name)" -id 4 -percentcomplete $($count/$tot*100)
                }
            }
        }
    }

    End {
    }
}
```

alagoutte commented:
> It should have been a separate pull.
Yes please !
> 
> I changed it so it uses write-verbose if you specify -verbose instead of having “noverbose” flag on it.
> 
Yes, it is not a good idea to have noverbose flag...

> And I also added so the name of the switch is added to the connection so if for instance you have 100 switches you can do
> $arubaswitchcollection|where name -eq 'switch1' or when adding vlans you get which switch you're adding it to.
> 
> I've made a small function that imports vlans from an excel file and adds them to a collection of switches. When doing this its preferable that errors etc refers to a specific switch. Or for instance see this code that I use for cleaning up a lab I was doing.
> 
> ```powershell
> function Remove-ArubaLabVLAN {
>     [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
>     Param(
>         [Parameter (Mandatory = $False)]
>         [ValidateNotNullOrEmpty()]
>         [PSObject]$connection = $DefaultArubaSWConnection
>     )
>     Begin {
>     }
> 
>     Process {
>         if ($null -eq $connection -or [string]::IsNullOrEmpty($connection)) {
>             write-error "No ArubaSwitch specified"
>             return
>         }
>         if ($connection.count -gt 1 -and $connection.GetType().basetype.name -eq 'Array') {
>             $buff = foreach ($aconn in $connection) {
>                 Remove-ArubaLabVLAN -connection $aconn
>             }
>             $buff
>         } else {
>             Write-Progress -activity "Remove Vlan on $($connection.Name)" -id 4 -percentcomplete 0
>             $tot = 8
>             foreach ($id in @(601..608)) {
>                 $count = $id-600
>                 $uri = "rest/v4/vlans/${id}"
>                 $target = "{0} on {1}" -f $id, $connection.Name
>                 if ($PSCmdlet.ShouldProcess($target, "Remove VLAN")) {
>                     Write-Progress -activity "Remove Vlan on $($connection.Name)" -status $id -id 4 -percentcomplete $($count/$tot*100)
>                     $null = Invoke-ArubaSWWebRequest -method "DELETE" -uri $uri -connection $connection
>                     Write-Progress -activity "Remove Vlan on $($connection.name)" -id 4 -percentcomplete $($count/$tot*100)
>                 }
>             }
>         }
>     }
> 
>     End {
>     }
> }
> ```
ok interesting...

alagoutte commented:
can you split your patch (and make a another PR for connection stuff ?)

blurpiflurp commented:
Yeah I'm working on a larger patch for more stuff so will do in a couple of days and we will see what you approve :)

alagoutte committed:
PSSA: Enable back ShouldProcess check

alagoutte committed:
Banner: use ShouldProcess

alagoutte committed:
DNS: Use ShouldProcess (for Set and Remove)

Only medium level for Set and high for Remove
add .EXAMPLE with -confirm:false

also update Test Suite

alagoutte committed:
LACP: Use ShouldProcess

alagoutte committed:
Led: Use ShouldProcess for Set (medium)

alagoutte committed:
LLDP: Use ShouldProcess for Set (medium)

alagoutte committed:
PoE: Use ShouldProcess for Set (medium)

alagoutte committed:
Port: Use ShouldProcess for Set (medium)

alagoutte committed:
RADIUSProfile: Use ShouldProcess for Set (medium)

alagoutte committed:
RADIUSServer: Use ShouldProcess (for Set and Remove)

Only medium level for Set and high for Remove

Update Also Test Suite

alagoutte committed:
RADIUSServerGroup: Use ShouldProcess (Remove)

high level for Remove

Update Also Test Suite

alagoutte committed:
RestTimeout: Use ShouldProcess for Set (medium)

alagoutte committed:
STP: Use ShouldProcess for Set (medium)

alagoutte committed:
System: Use ShouldProcess for Set (medium)

alagoutte committed:
Trunk: Use ShouldProcess (Remove)

high level for Remove

Update Also Test Suite

alagoutte committed:
Vlans-Ports: Use ShouldProcess (for Set and Remove)

Only medium level for Set and high for Remove

Update Also Test Suite

alagoutte committed:
Vlans: Use ShouldProcess for Set (medium)

alagoutte committed:
SSl(private): Ignore PSSA ShouldProcess Warning

alagoutte committed:
Cookie: Ignore PSSA ShouldProcess (and use standard param())

alagoutte committed:
Vlans: Remove Write-Progress and set SupportsShouldProcess before Param()

alagoutte committed:
(Dis)Connect: Use ShouldProcess and update Test Suite

alagoutte committed:
Ping(Tests): Fix test (noconfirm -> confirm for DNS)

alagoutte committed:
banner: add missing target (set to connection server)

alagoutte committed:
LLDP: add missing target (set to connection server)

alagoutte committed:
Port: add missing target (set to port_id)

alagoutte committed:
RADIUSProfile: add missing target (set to connection server)

alagoutte committed:
RestTimeout: add missing target (set to connection server)

alagoutte committed:
Led: Remove extra parenthese
